### PR TITLE
add a share dialog

### PR DESCRIPTION
**Pull request by pythongiant**
add a share dialog

add icon

issues #1779

Changes: Add a share modal 

Demo Link: https://pr-1807-fossasia-susi-web-chat.surge.sh

Screenshots for the change: 
image

### CODE REVIEW

**Comment by pythongiant** [comment]
Yeah but it would be easier for the user to do it from chat page @akshatnid I feel that's better user experience

**Comment by utkarsh2102** [comment]
+1 to @pythongiant.
Been using it. And it would be a good feature, IMHO. 
Enhances the accessibility. I think. It deserves to be added, @akshatnitd. What's your say?

**Comment by pythongiant** [comment]
@rajvaibhavdubey can you check it out now it should be fixed.
@akshatnitd  can we merge this now

**Comment by AakashMallik** [comment]
screenshot-codepen io-2018 12 11-15-27-54

Please use this UI design instead. The one we have doesn't look good.

**Comment by pythongiant** [comment]
How about now. @niteshkumarniranjan you don't have to make a new PR now. I tried to implement your design in my own way. @AakashMallik Looks Good now?
image

**Comment by niteshkumarniranjan** [comment]
@pythongiant There is still a big difference in the my implementation and yours.

**Comment by AakashMallik** [comment]
@pythongiant Be honest, are you satisfied with how it looks? If not, keep changing it. The current implementation still looks different from the image I posted. Keep up the good work.

**Comment by pythongiant** [comment]
@akshatnitd check it out now?

**Comment by AakashMallik** [comment]
@pythongiant I just gave you the raw code. Move the styles into a style object as we have done for other objects.

File: src/components/ChatApp/MessageSection/DialogSection.js [hunk 1]
```diff
@@ -1,21 +1,88 @@
...
+      flexDirection: 'column',
+      marginTop: '8px',
+    },
+    IconTitleStyle: {
+      height: '48px',
+      lineHeight: '48px',
+      verticalAlign: 'center',
+      marginLeft: '32px',
+    },
+    IconParent: {
```
Comment (AakashMallik) [inline src/components/ChatApp/MessageSection/DialogSection.js]: Change name to IconWrapperStyle

File: src/components/ChatApp/MessageSection/DialogSection.js [hunk 2]
```diff
@@ -1,21 +1,88 @@
...
+        contentStyle={{
+          width: '350px',
+          minWidth: '350px',
+          textAlign: 'center',
+        }}
+        open={props.openShare}
+        onRequestClose={props.handleShareClose}
+      >
+        <div
+          style={{
```
Comment (AakashMallik) [inline src/components/ChatApp/MessageSection/DialogSection.js]: Enclose this in object and name it ShareTitleWrapperStyle

File: src/components/ChatApp/MessageSection/DialogSection.js [hunk 3]
```diff
@@ -1,21 +1,88 @@
...
+      right: '10px',
+      top: '10px',
+      cursor: 'pointer',
+    },
   };
   return (
     <div>
+      {/* Share */}
+      <Dialog
+        contentStyle={{
```
Comment (AakashMallik) [inline src/components/ChatApp/MessageSection/DialogSection.js]: Enclose and name ShareDialogContainerStyle

File: src/components/ChatApp/MessageSection/DialogSection.js [hunk 4]
```diff
@@ -1,21 +1,88 @@
...
-    fill: '#444',
-    width: '26px',
-    height: '26px',
-    right: '10px',
-    top: '10px',
-    cursor: 'pointer',
+  const shareUrl = 'http://chat.susi.ai';
+  const title =
+    'Susi is an artificial intelligence system, combining pattern matching, internet data, data flow-, and inference engine principles. Through some abilities to reflect, it can remember the user input to produce deductions and personalized feedback. Its purpose is to explore the abilities of an artificial companion and to answer the remaining unanswered questions. The SUSI.AI web chat is a front-end developed for web access of SUSI.';
+  const style = {
```
Comment (AakashMallik) [inline src/components/ChatApp/MessageSection/DialogSection.js]: Move this outside the component declaration. ie. Place it between line number 10 and 11

Comment (AakashMallik) [reply src/components/ChatApp/MessageSection/DialogSection.js]: Also, name it style**s**. AFAIK, we have used styles in other components.

File: src/components/ChatApp/MessageSection/DialogSection.js [hunk 5]
```diff
@@ -13,9 +34,55 @@ const DialogSection = props => {
...
+        onRequestClose={props.handleShareClose}
+      >
+        <div style={styles.titleWrapperStyle}>
+          <h3>Share about SUSI</h3>
+          <div style={{ flex: 1 }} />
+          <IconButton onTouchTap={props.handleShareClose}>
+            <CloseIcon size={32} />
+          </IconButton>
+        </div>
+        <div style={styles.ShowParent}>
```
Comment (AakashMallik) [inline src/components/ChatApp/MessageSection/DialogSection.js]: Change this shareIconsContainer

File: src/components/ChatApp/MessageSection/DialogSection.js [hunk 6]
```diff
@@ -13,9 +34,55 @@ const DialogSection = props => {
...
+    width: '350px',
+    minWidth: '350px',
+    textAlign: 'center',
+  },
+  titleWrapperStyle: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+};
+const DialogSection = props => {
```
Comment (AakashMallik) [inline src/components/ChatApp/MessageSection/DialogSection.js]: new line after line 47

File: src/components/ChatApp/MessageSection/DialogSection.js [hunk 7]
```diff
@@ -1,10 +1,31 @@
...
+import IconButton from 'material-ui/IconButton';
+import CloseIcon from 'material-ui/svg-icons/navigation/close';
+const { FacebookShareButton, TwitterShareButton } = ShareButtons;
+const FacebookIcon = generateShareIcon('facebook');
+const TwitterIcon = generateShareIcon('twitter');
 import Close from 'material-ui/svg-icons/navigation/close';
-
-const DialogSection = props => {
-  const closingStyle = {
+const styles = {
```
Comment (AakashMallik) [inline src/components/ChatApp/MessageSection/DialogSection.js]: new line after line 10

**Comment by AakashMallik** [comment]
Good job. @pythongiant

**Comment by pythongiant** [comment]
@saarthakchats  it is not responsive

**Comment by saarthakchats** [comment]
Why does that make a difference, what I mean is that for following your link one must click on the icon; make it work when clicking on the texts - 'Facebook', 'Twitter' - as well.

**Comment by pythongiant** [comment]
Well it does matter cause the react component we use has diffrent styles associated with it for some reason
> Why does that make a difference, what I mean is that for following your link one must click on the icon; make it work when clicking on the texts - 'Facebook', 'Twitter' - as well.

**Comment by AakashMallik** [comment]
@pythongiant Whatever @saarthakchats has suggested is viable and possible. Make it happen. Also, make it responsive.

**Comment by pythongiant** [comment]
@saarthakchats @AakashMallik check it out now. should work

**Comment by pythongiant** [comment]
@niteshkumarniranjan if you could review it now.
BTW is this really necessary?

**Comment by pythongiant** [comment]
@akshatnitd @mariobehling

**Comment by akshatnitd** [comment]
@pythongiant Thanks for your contribution :)

I would like to mention that, from next time, please take consent from mentors before moving ahead with a feature. Bug Fixes can be reported and solved, but feature requests should be approved first.

**Comment by pythongiant** [comment]
oh ok cool. will keep that in mind @akshatnitd

**Pull request merged**